### ISSUE TITLE
Error en el módulo webtrees-pedigree-chart 1.4.0

### ISSUE THREAD

**Issue by BernatBanyuls** [magicsunday/webtrees-pedigree-chart#53]
I get the following error when requesting the graphic. I am using webtrees 2.1.7 and I have tried to upgrade to module version 1.4.0, but I still get the same error. Can you help me? Thank you.

DOMDocument::loadHTML(): Unexpected end tag : span in Entity, line: 1 …/modules_v4/webtrees-pedigree-chart/src/Traits/IndividualTrait.php:128
#0 [internal function]: Fisharebest\Webtrees\Webtrees::Fisharebest\Webtrees\{closure}(2, 'DOMDocument::lo...', '/var/www/vhosts...', 128, Array)
#1 …/modules_v4/webtrees-pedigree-chart/src/Traits/IndividualTrait.php(128): DOMDocument->loadHTML('<span class="NA...')
#2 …/modules_v4/webtrees-pedigree-chart/src/Traits/IndividualTrait.php(79): MagicSunday\Webtrees\PedigreeChart\Module->getXPath('<span class="NA...')
#3 …/modules_v4/webtrees-pedigree-chart/src/Module.php(248): MagicSunday\Webtrees\PedigreeChart\Module->getIndividualData(Object(Cissee\WebtreesExt\IndividualExt), 3)
#4 …/modules_v4/webtrees-pedigree-chart/src/Module.php(258): MagicSunday\Webtrees\PedigreeChart\Module->buildJsonTree(Object(Cissee\WebtreesExt\IndividualExt), 3)
#5 …/modules_v4/webtrees-pedigree-chart/src/Module.php(258): MagicSunday\Webtrees\PedigreeChart\Module->buildJsonTree(Object(Cissee\WebtreesExt\IndividualExt), 2)
#6 …/modules_v4/webtrees-pedigree-chart/src/Module.php(172): MagicSunday\Webtrees\PedigreeChart\Module->buildJsonTree(Object(Cissee\WebtreesExt\IndividualExt))
#7 …/app/Http/Middleware/RequestHandler.php(54): MagicSunday\Webtrees\PedigreeChart\Module->handle(Object(Nyholm\Psr7\ServerRequest))
#8 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\RequestHandler->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#9 …/modules_v4/reminder/src/Modules/ReminderModule.php(97): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#10 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): UksusoFF\WebtreesModules\Reminder\Modules\ReminderModule->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#11 …/modules_v4/vesta_research_suggestions/ResearchSuggestionsModule.php(277): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#12 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Cissee\Webtrees\Module\ResearchSuggestions\ResearchSuggestionsModule->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#13 …/modules_v4/faces/src/Modules/FacesModule.php(94): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#14 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): UksusoFF\WebtreesModules\Faces\Modules\FacesModule->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#15 …/app/Module/HitCountFooterModule.php(154): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#16 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Module\HitCountFooterModule->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#17 …/app/Module/CheckForNewVersion.php(115): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#18 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Module\CheckForNewVersion->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#19 …/app/Http/Middleware/CheckCsrf.php(80): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#20 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CheckCsrf->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#21 …/vendor/oscarotero/middleland/src/Dispatcher.php(118): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#22 …/app/Webtrees.php(275): Middleland\Dispatcher->dispatch(Object(Nyholm\Psr7\ServerRequest))
#23 …/app/Http/Middleware/Router.php(153): Fisharebest\Webtrees\Webtrees::dispatch(Object(Nyholm\Psr7\ServerRequest), Array)
#24 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\Router->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#25 …/app/Http/Middleware/BootModules.php(60): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#26 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\BootModules->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#27 …/app/Http/Middleware/RegisterGedcomTags.php(54): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#28 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\RegisterGedcomTags->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#29 …/app/Http/Middleware/LoadRoutes.php(75): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#30 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\LoadRoutes->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#31 …/app/Http/Middleware/CheckForNewVersion.php(65): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#32 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CheckForNewVersion->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#33 …/app/Http/Middleware/UseTransaction.php(45): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#34 …/vendor/illuminate/database/Concerns/ManagesTransactions.php(29): Fisharebest\Webtrees\Http\Middleware\UseTransaction::Fisharebest\Webtrees\Http\Middleware\{closure}(Object(Illuminate\Database\MySqlConnection))
#35 …/app/Http/Middleware/UseTransaction.php(46): Illuminate\Database\Connection->transaction(Object(Closure), 3)
#36 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseTransaction->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#37 …/app/Http/Middleware/DoHousekeeping.php(73): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#38 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\DoHousekeeping->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#39 …/app/Http/Middleware/UseTheme.php(69): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#40 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseTheme->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#41 …/app/Http/Middleware/CheckForMaintenanceMode.php(51): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#42 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CheckForMaintenanceMode->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#43 …/app/Http/Middleware/UseLanguage.php(71): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#44 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseLanguage->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#45 …/app/Http/Middleware/UseSession.php(78): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#46 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseSession->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#47 …/app/Http/Middleware/UpdateDatabaseSchema.php(57): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#48 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UpdateDatabaseSchema->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#49 …/app/Http/Middleware/UseDatabase.php(118): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#50 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\UseDatabase->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#51 …/app/Http/Middleware/BadBotBlocker.php(233): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#52 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\BadBotBlocker->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#53 …/app/Http/Middleware/CompressResponse.php(73): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#54 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\CompressResponse->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#55 …/app/Http/Middleware/ContentLength.php(40): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#56 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ContentLength->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#57 …/vendor/middlewares/client-ip/src/ClientIp.php(65): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#58 …/app/Http/Middleware/ClientIp.php(47): Middlewares\ClientIp->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#59 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ClientIp->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#60 …/app/Http/Middleware/HandleExceptions.php(90): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#61 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\HandleExceptions->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#62 …/app/Http/Middleware/BaseUrl.php(79): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#63 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\BaseUrl->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#64 …/app/Http/Middleware/ReadConfigIni.php(68): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#65 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\ReadConfigIni->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#66 …/app/Http/Middleware/SecurityHeaders.php(48): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#67 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\SecurityHeaders->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#68 …/app/Http/Middleware/EmitResponse.php(57): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#69 …/vendor/oscarotero/middleland/src/Dispatcher.php(136): Fisharebest\Webtrees\Http\Middleware\EmitResponse->process(Object(Nyholm\Psr7\ServerRequest), Object(Middleland\Dispatcher))
#70 …/vendor/oscarotero/middleland/src/Dispatcher.php(118): Middleland\Dispatcher->handle(Object(Nyholm\Psr7\ServerRequest))
#71 …/app/Webtrees.php(275): Middleland\Dispatcher->dispatch(Object(Nyholm\Psr7\ServerRequest))
#72 …/app/Webtrees.php(262): Fisharebest\Webtrees\Webtrees::dispatch(Object(Nyholm\Psr7\ServerRequest), Array)
#73 …/index.php(51): Fisharebest\Webtrees\Webtrees->httpRequest()
#74 {main}

**Comment by magicsunday** [comment]
Apologies for the late reply. Were you able to fix the problem in the meantime? Have you tried a more recent version of Webtrees, or the state from the extension's master branch? Otherwise I need further information, please always use the template for the error message if possible. It might also be helpful if you could provide me with a sample data set that I can use to reproduce the problem.

**Comment by BernatBanyuls** [comment]
Gracias por tu interés.
Vamos a ver...
Hoy he vuelto a activar el módulo y he realizado diferentes pruebas para buscar el error.

En primer lugar tienes una imagen de los ficheros de pedigree-char en mi servidor

![image](https://user-images.githubusercontent.com/103060233/211766410-5460b400-a025-433b-8aaf-3bc7ee69d05c.png)

En esta siguiente imagen tienes todos los módulos que utilizo actualmente

![image](https://user-images.githubusercontent.com/103060233/211766796-d1e3ad06-26ab-40e6-aa0d-c4be282e8989.png)

Y en esta tercera captura, el error que me devuelve webtrees

![image](https://user-images.githubusercontent.com/103060233/211767866-5358d236-4f79-4975-98a1-70c39e048fa7.png)

De todos modos, aquí expongo algunas "particularidades".

- El error se produce en "algunos" registros. No en todos. Ejemplo: En el mío propio y en el de mi hijo. En el de mi mujer y antepasados, no.
- Cuando se produce el error, "disminuyo el número de generaciones" y funciona. (Hasta 3 generaciones no hay problema. A partir de "4" se produce el error). En otros registros he probado con "5" y "6" y funciona correctamente.
- -Sigo experimentando y generalmente suelen salir los errores en miembros de mi família. ¿Tendré ago corrupto? Con las pruebas que estoy realizando con otros registros todo parece funcionar correctamente.
- 
En fin, creo que lo que más te puede orientar es que el error me sale cuando aumento "generaciones".  Si le pido  gráficos de 4 generaciones y me da error, lo bajo a 3 o a 2 hasta que desaparece el problema. Cuando vuelvo a aumentar las generaciones, el problema vuelve.

Espero haberme explicado un poco mejor y a ver si esto te ayuda.

Gracias.
-

**Comment by magicsunday** [comment]
That almost sounds to me as if there might be a problem with the spelling and the processing of one or more names. Can you please send me a gedcom file with a small part of your family tree (which is causing the error).

**Comment by BernatBanyuls** [comment]
Tengo el GEDCOM pero no puedo mandarlo por aquí (?) Puedo enviarlo por email?

**Comment by magicsunday** [comment]
Please send ist to magic_sunday@gmx.de

**Comment by magicsunday** [comment]
Very Strange. I do not get any errors. What version of webtrees are you using? What PHP version? Does the gedcom file you send me, produces the error on your installation? Does the website log, in the control panel, show any additional information?

It would be helpful to see the full line of code which causes the error in the line 128 `DOMDocument->loadHTML('<span class="NA...')` in file IndividualTrait.php. Is it possible for you to add a debug statement to the line before? 

```
var_dump($fullname);
var_dump(mb_convert_encoding($fullName, 'HTML-ENTITIES', 'UTF-8'));
```

**Comment by magicsunday** [comment]
I looked up your list of custom modules installed and found out, that the `vesta_classic_look_and_feel` causes the issues. Especially the option "Display nicknames before surnames", which is the cause of the error. After disabling it the chart renders perfectly.

**Comment by BernatBanyuls** [comment]
Perfecto, Rico!

 

Muchas gracias por tu paciencia. Tendré en cuenta tu nota para el futuro.

 

Gracias.

 

PD.

Port cierto, es fácil y posible poder cambiar el nombre a “pedigree chart”? Como ves, en el menú (al catalán)  me aparecen las dos opciones con el mismo nombre.



 

En “Español” no hay problema



 

De todos modos, esto es poco importante. 



Saludos.

 

Abans d'imprimir aquest correu, penseu si és necessari.

Aquest correu electrònic pot contenir informació confidencial o legalment protegida i està exclusivament adreçat a la persona o entitat destinatària. Si no sou el destinatari final o la persona encarregada de recollir-lo, no esteu autoritzat a llegir-lo, retenir-lo, modificar-lo, distribuir-lo, copiar-lo ni a revelar-ne el contingut. Si heu rebut aquest correu electrònic per error, us preguem que informeu el remitent i elimineu del vostre sistema el missatge i el material annex que pugui contenir. 
Gràcies per la vostra col·laboració.

 

De: Rico Sonntag ***@***.***> 
Enviat: dissabte, 14 de gener de 2023 16:51
Per a: magicsunday/webtrees-pedigree-chart ***@***.***>
A/c: BernatBanyuls ***@***.***>; State change ***@***.***>
Tema: Re: [magicsunday/webtrees-pedigree-chart] Error en el módulo webtrees-pedigree-chart 1.4.0 (Issue #53)

 

I looked up your list of custom modules installed and found out, that the vesta_classic_look_and_feel causes the issues. Especially the option "Display nicknames before surnames", which is the cause of the error. After disabling it the chart renders perfectly.

**Comment by BernatBanyuls** [comment]
Perfecto!

 

He eliminado la “barra inclinada” y todo funciona perfectamente.

 

No creo que tenga muchos registros con este símbolo. Los buscaré, pero ahora todo funciona correctamente.

 

Te pido disculpas por tu tiempo, pero es que mi formación informática no me permitía investigarlo por mi solo.

 

Muchas Gracias.

 

Abans d'imprimir aquest correu, penseu si és necessari.

Aquest correu electrònic pot contenir informació confidencial o legalment protegida i està exclusivament adreçat a la persona o entitat destinatària. Si no sou el destinatari final o la persona encarregada de recollir-lo, no esteu autoritzat a llegir-lo, retenir-lo, modificar-lo, distribuir-lo, copiar-lo ni a revelar-ne el contingut. Si heu rebut aquest correu electrònic per error, us preguem que informeu el remitent i elimineu del vostre sistema el missatge i el material annex que pugui contenir. 
Gràcies per la vostra col·laboració.

 

De: Rico Sonntag ***@***.***> 
Enviat: dissabte, 14 de gener de 2023 16:51
Per a: magicsunday/webtrees-pedigree-chart ***@***.***>
A/c: BernatBanyuls ***@***.***>; State change ***@***.***>
Tema: Re: [magicsunday/webtrees-pedigree-chart] Error en el módulo webtrees-pedigree-chart 1.4.0 (Issue #53)

 

I looked up your list of custom modules installed and found out, that the vesta_classic_look_and_feel causes the issues. Especially the option "Display nicknames before surnames", which is the cause of the error. After disabling it the chart renders perfectly.

**Comment by ric2016** [comment]
Note: Fixed in Vesta release 2.1.16.0.0.
